ubuntu_image/image.py

```diff
@@ -45,6 +50,17 @@ def __init__(self, path, size):
         # will cause all the bytes to read as zero.  Stevens $4.13
         os.truncate(path, 0)
         os.truncate(path, size)
+        # Prepare the device and disk objects for parted to be used for all
+        # future partition() calls.  Only do it if we actually care about the
+        # partition table.
+        if schema:
+            self.device = parted.Device(self.path)
+            label = 'msdos' if schema is VolumeSchema.mbr else 'gpt'
+            self.schema = schema
+            self.disk = parted.freshDisk(self.device, label)
+            self.sector_size = self.device.sectorSize
+        else:
+            self.sector_size = 512
```

Comment:
The conditional should be `if schema is not None` but then really we should swap the if and else clauses (since positive tests are preferred to negative tests).  So:

```python

        if schema is None:
            self.sector_size = 512
        else:
            self.device = parted.Device(self.path)
            label = 'msdos' if schema is VolumeSchema.mbr else 'gpt'
            self.schema = schema
            self.disk = parted.freshDisk(self.device, label)
            self.sector_size = self.device.sectorSize
```